File: Principal/Html/Componente/ArquivoTile.cs
using NetZ.Web.Html;
using NetZ.Web.Html.Componente.Mobile;
using NetZ.Web.Server.Arquivo.Css;

namespace Drive.Html.Componente
{
    internal class ArquivoTile : TileBase
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private Div _divDttAlteracao;
        private Div _divIcone;
        private Div _divMenu;
        private Div _divTitulo;

        private Div divDttAlteracao
        {
            get
            {
                if (_divDttAlteracao != null)
                {
                    return _divDttAlteracao;
                }

                _divDttAlteracao = new Div();

                return _divDttAlteracao;
            }
        }

        private Div divIcone
        {
            get
            {
                if (_divIcone != null)
                {
                    return _divIcone;
                }

                _divIcone = new Div();

                return _divIcone;
            }
        }

        private Div divMenu
        {
            get
            {
                if (_divMenu != null)
                {
                    return _divMenu;
                }

                _divMenu = new Div();

                return _divMenu;
            }
        }

        private Div divTitulo
        {
            get
            {
                if (_divTitulo != null)
                {
                    return _divTitulo;
                }

                _divTitulo = new Div();

                return _divTitulo;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override bool getBooJs()
        {
            return true;
        }

        protected override void inicializar()
        {
            base.inicializar();

            this.divDttAlteracao.strConteudo = "_dtt_alteracao_conteudo";
            this.divDttAlteracao.strId = "_dtt_alteracao_id";

            this.divIcone.strId = "_icone_id";

            this.divTitulo.strConteudo = "_arquivo_titulo";
            this.divTitulo.strId = "_titulo_id";

            this.strId = "_tile_id";
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            this.divIcone.setPai(this);
            this.divTitulo.setPai(this);
            this.divDttAlteracao.setPai(this);
            this.divMenu.setPai(this);
        }

        protected override void setCss(CssArquivo css)
        {
            base.setCss(css);

            this.addCss(css.setBoxShadow(0, 1, 5, 0));
            this.addCss(css.setCursor("pointer"));
            this.addCss(css.setDisplay("none"));
            this.addCss(css.setHeight(50));
            this.addCss(css.setMargin(10));
            this.addCss(css.setPosition("relative"));

            this.divDttAlteracao.addCss(css.setColor("grey"));
            this.divDttAlteracao.addCss(css.setFontSize(12));

            this.divIcone.addCss(css.setBackgroundImage("http://megaicons.net/static/img/icons_sizes/8/178/256/very-basic-file-icon.png"));
            this.divIcone.addCss(css.setBackgroundPosition("center"));
            this.divIcone.addCss(css.setBackgroundRepeat("no-repeat"));
            this.divIcone.addCss(css.setBackgroundSize("30px 30px"));
            this.divIcone.addCss(css.setFloat("left"));
            this.divIcone.addCss(css.setHeight(100, "%"));
            this.divIcone.addCss(css.setMarginRight(10));
            this.divIcone.addCss(css.setWidth(50));

            this.divMenu.addCss(css.setBackgroundImage("https://cdn0.iconfinder.com/data/icons/pack-web-app-game/512/option-menu-256.png"));
            this.divMenu.addCss(css.setBackgroundPosition("center"));
            this.divMenu.addCss(css.setBackgroundRepeat("no-repeat"));
            this.divMenu.addCss(css.setBackgroundSize("25px"));
            this.divMenu.addCss(css.setHeight(100, "%"));
            this.divMenu.addCss(css.setPosition("absolute"));
            this.divMenu.addCss(css.setRight(0));
            this.divMenu.addCss(css.setTop(0));
            this.divMenu.addCss(css.setWidth(35));

            this.divTitulo.addCss(css.setPaddingTop(5));
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: Drive.TypeScript/html/componente/ArquivoTile.ts
/// <reference path="../../../Web.TypeScript/erro/Erro.ts"/>
/// <reference path="../../../Web.TypeScript/html/componente/mobile/TileBase.ts"/>
/// <reference path="../../../Web.TypeScript/html/Div.ts"/>
/// <reference path="../../../Web.TypeScript/OnClickListener.ts"/>
/// <reference path="../../../Web.TypeScript/Utils.ts"/>

module Drive
{
    // #region Importações

    import Div = Web.Div;
    import Erro = Web.Erro;
    import OnClickListener = Web.OnClickListener;
    import TileBase = Web.TileBase;
    import Utils = Web.Utils;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class ArquivoTile extends TileBase implements OnClickListener
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private _arq: ArquivoDominio;
        private _divDttAlteracao: Div;
        private _divIcone: Div;
        private _divTitulo: Div;

        private get arq(): ArquivoDominio
        {
            return this._arq;
        }

        private set arq(arq: ArquivoDominio)
        {
            this._arq = arq;
        }

        private get divDttAlteracao(): Div
        {
            if (this._divDttAlteracao != null)
            {
                return this._divDttAlteracao;
            }

            this._divDttAlteracao = new Div(this.strId + "_divDttAlteracao");

            return this._divDttAlteracao;
        }

        private get divIcone(): Div
        {
            if (this._divIcone != null)
            {
                return this._divIcone;
            }

            this._divIcone = new Div(this.strId + "_divIcone");

            return this._divIcone;
        }

        private get divTitulo(): Div
        {
            if (this._divTitulo != null)
            {
                return this._divTitulo;
            }

            this._divTitulo = new Div(this.strId + "_divTitulo");

            return this._divTitulo;
        }

        // #endregion Atributos

        // #region Construtores

        constructor(intIndex: number, arq: ArquivoDominio)
        {
            super(ArquivoTile.name + "_" + intIndex.toString());

            this.arq = arq;
        }

        // #endregion Construtores

        // #region Métodos

        private abrir(): void
        {
            if (this.arq == null)
            {
                return;
            }

            if (this.arq.booPasta)
            {
                this.abrirConteudo();
            }
            else
            {
                this.abrirDetalhe();
            }
        }

        private abrirConteudo(): void
        {
            PagPrincipalDrive.i.abrirConteudo(this.arq);
        }

        private abrirDetalhe(): void
        {
            PagPrincipalDrive.i.abrirDetalhe(this.arq);
        }

        protected inicializar(): void
        {
            super.inicializar();

            this.inicializarDivIcone();

            this.mostrar();
        }

        private inicializarDivIcone(): void
        {
            if (this.arq == null)
            {
                return;
            }

            if (this.arq.booPasta)
            {
                this.divIcone.jq.css("background-image", "url(https://cdn2.iconfinder.com/data/icons/metro-uinvert-dock/128/Folder_-_Google_Docs.png)");
            }
        }

        protected montarLayoutFixo(strLayoutFixo: string): string
        {
            strLayoutFixo = super.montarLayoutFixo(strLayoutFixo);

            if (Utils.getBooStrVazia(strLayoutFixo))
            {
                return null;
            }

            strLayoutFixo = strLayoutFixo.replace("_tile_id", this.strId);

            strLayoutFixo = strLayoutFixo.replace("_dtt_alteracao_conteudo", Utils.getStrDttAmigavel(this.arq.dttAlteracao));
            strLayoutFixo = strLayoutFixo.replace("_dtt_alteracao_id", this.divDttAlteracao.strId);

            strLayoutFixo = strLayoutFixo.replace("_icone_id", this.divIcone.strId);

            strLayoutFixo = strLayoutFixo.replace("_arquivo_titulo", this.arq.strNome);
            strLayoutFixo = strLayoutFixo.replace("_titulo_id", this.divTitulo.strId);

            return strLayoutFixo;
        }

        protected setEventos(): void
        {
            super.setEventos();

            this.addEvtOnClickListener(this);
        }

        // #endregion Métodos

        // #region Eventos

        public onClick(objSender: Object, arg: JQueryEventObject): void
        {
            try
            {
                switch (objSender)
                {
                    case this:
                        this.abrir();
                        return;
                }
            }
            catch (ex)
            {
                new Erro("Erro desconhecido.", ex);
            }
        }

        // #endregion Eventos
    }
}